imageio/imageio-psd/src/test/java/com/twelvemonkeys/imageio/plugins/psd/PSDImageReaderTest.java

```diff
@@ -532,4 +534,41 @@ public void testReadUnicodeLayerName() throws IOException {
             assertEquals("If_The_Layer_Name_Is_Really_Long_Oh_No_What_Do_I_Do", ((IIOMetadataNode) layerInfo.item(0)).getAttribute("name"));
         }
     }
+
+    @Test
+    public void testGroupLayerRead() throws IOException {
+        PSDImageReader imageReader = createReader();
+
+        try (ImageInputStream stream = ImageIO.createImageInputStream(getClassLoaderResource("/psd/group_layer.psd"))) {
+            imageReader.setInput(stream);
+
+            IIOMetadata metadata = imageReader.getImageMetadata(0);
+
+            List<PSDLayerInfo> layerInfos = ((PSDMetadata) metadata).layerInfo;
+
```

Comment:
Another thing: You need to assert that this list is indeed 3 items and contains the ids 40, 42 and 43, otherwise the loop below will simply pass with nothing being tested. Which is probably not the intention. 😀